frontend/src/components/GameMap/Activity.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import { Square, ActivityImg } from './ActivityStyles';
+import ExpeditionImg from './resources/expedition.png';
+import InformationImg from './resources/information.png';
+import SurveyImg from './resources/survey.png';
+import TaskImg from './resources/task.png';
+
+
+function Activity(props) {
+    const [width, setWidth] = useState(100);
+    const [height, setHeight] = useState(100);
+
+    useEffect(() => {
+        const updateSize = () => {
+            setWidth(props.parent.current.offsetWidth / props.mapSizeY * 0.90); // * 0.90 prevents wraping row
+            setHeight(props.parent.current.offsetHeight / props.mapSizeX * 0.90);
+        }
+
+        if (props.parent.current) 
+        {
+            updateSize();
+            window.addEventListener("resize", () => {
+                updateSize();
+            }, false);
+        }
+        
+    }, [width, height, props])
+
+    const isUnlocked = () => {
+        // TODO
+        return true;
+    }
+
+    const getImg = (type) => {
+        switch(type) {
+            case "expedition":
+                return ExpeditionImg
+            case "information":
+                return InformationImg;
+            case "survey":
+                return SurveyImg;
+            case "task":
+                return TaskImg;
+            default:
+                return;
+          }
+
+    }
+
+    const getContent = () => {
+        if (props.props)
+        {
+            if (!isUnlocked())
+            {
+                return;
+            }
+            const activity = props.props;
+            const type = activity.activityType;
+
+            return <ActivityImg src={getImg(type)}></ActivityImg>
```

Comment:
Nie wiem czy dobrze to rozumiem, ale kod pod if'em nie wykona się jeżeli `isUnlocked() === false`. Nie zadziała więc zapisanie tego w ten sposób?
```suggestion
            if (isUnlocked()){
                const activity = props.props;
                const type = activity.activityType;
                return <ActivityImg src={getImg(type)}></ActivityImg>
            }
```
No chyba, że źle coś zrozumiałem albo w tym if'ie coś się w przyszłości wydarzy